datalad_registry/models.py

```diff
@@ -25,6 +25,9 @@ class URL(db.Model):  # type: ignore
     #: Whether initial data has been collected for this URL
     processed = db.Column(db.Boolean, default=False, nullable=False)
 
+    # The path in the local cache where a copy of the dataset at the URL is stored
+    cache_path = db.Column(db.String(34), nullable=True, default=None)
```

Comment:
So what this `cache_path` be now?  Please describe briefly in a comment above, so it also becomes clear why `34` is a sufficient length.

Reply:
`cache_path` will be a relative path of 

```python
        uuid = uuid4().hex
        path = Path(uuid[:3], uuid[3:6], uuid[6:])
```

The path relative to the directory specified by `DATALAD_REGISTRY_DATASET_CACHE`, the base cache path.

Reply:
Great, thank you for the explanation!

We need an an endpoint which , given a `cache_path`  would return a dataset record. And then also a command line entry point for the same, and ideally it should be smart to operate on full path, thus taking the last three path components -- this way user (us) could quickly figure out what dataset a given path corresponds to.

Reply:
It makes complete sense to me to have such a command line entry for us, the operators, to have access to such information conveniently. However, since `cache_path` is of the internal organization of a running instance of datalad-registry. Don't you think it is a bad idea to expose it through the public `API`?

Reply:
I don't see a problem with that, in particular if it is just a UUID (random) and you need to know it to get some non 404 response.  What concerns do you have from exposing it?

Reply:
Not particular concern in mind. I just prefer withholding as much internal information as possible from the public as possible.

As an operator, you will, possible always, have access to the command line. Exposing the mapping of dataset URL records and `cache_path` through an API endpoint provides an additional means of access for convenience.

Let me know your final decision on whether you want the command line support alone or both the command line and API endpoint support. I will file an issue correspondingly.

Reply:
target use -- command line. I just didn't want to create another app which would interface to the same DB in parallel to the API app.  But if you think it is easier/ok -- go ahead without API component for now.

Reply:
A new issue is created for this feature. https://github.com/datalad/datalad-registry/issues/165